test/controllers/api/submissions_controller_test.rb

```diff
@@ -0,0 +1,87 @@
+require_relative './base_test_case'
+
+class API::SubmissionsControllerTest < API::BaseTestCase
+  ###
+  # CREATE
+  ###
+  test "create should return 401 with incorrect token" do
+    post api_solution_submissions_path(1), headers: @headers, as: :json
+    assert_response 401
+  end
+
+  test "create should 404 if the solution doesn't exist" do
+    setup_user
+    post api_solution_submissions_path(999), headers: @headers, as: :json
+    assert_response 404
+  end
+
+  test "create should 404 if the solution belongs to someone else" do
+    setup_user
+    solution = create :concept_solution
+    post api_solution_submissions_path(solution.uuid), headers: @headers, as: :json
+    assert_response 403
+    expected = { error: {
+      type: "solution_not_accessible",
+      message: I18n.t('api.errors.solution_not_accessible')
```

Comment:
This looks like an implementation detail. Maybe just have the error message itself here? It does make the test a bit more brittle, but it doesn't tie the test into any implementation detail.

Reply:
Yeah - that's an interesting point. So right now we do this for all the API tests, so I'll keep this to be consistent, but we should consider changing it everywhere. There's pros and cons to it.

Reply:
Absolutely